magento/pwa-studio
# [Doc] buildpack ref docs

## Description

This PR updates the JSDocs in the targets in the Buildpack package.
It also creates a file in devdocs that pulls this information into a reference-style topic.

## Related Issue

Closes PWA-703

## Acceptance 

@zetlen 

### Verification Stakeholders

@zetlen 

### Specification



### Verification Steps

1. Navigate to the pwa devdocs directory: `cd pwa-devdocs`
1. Start the HTML preview server: `yarn develop`
1. Verify files are auto-generated:

    - `src/_includes/auto-generated/pwa-buildpack/lib/WebpackTools/ModuleTransformConfig.md`
    - `src/_includes/auto-generated/pwa-buildpack/lib/BuildBus/declare-base.md`
    - `src/_includes/auto-generated/pwa-buildpack/lib/BuildBus/BuildBus.md`
    - `src/_includes/auto-generated/pwa-buildpack/lib/BuildBus/TargetProvider.md`
    - `src/_includes/auto-generated/pwa-buildpack/lib/BuildBus/Target.md`
    - `src/_includes/auto-generated/pwa-buildpack/lib/Utilities/getEnvVarDefinitions.js`

1. On the devdocs preview site, in the top navigation, click on **PWA Buildpack** under the _Reference API_ item.
1. Verify new navigation layout
1. Verify all the links under the Extension points, Extension framework, Developer utilities, Webpack tools, and Environment configuration sections work.

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
